scripts/pylib/pytest-twister-harness/src/twister_harness/device/hardware_adapter.py

```diff
@@ -170,6 +170,7 @@ def _disconnect_device(self) -> None:
             self._serial_connection.close()
             # self._serial_connection = None
             logger.debug('Closed serial connection for %s', serial_name)
+        self._serial_buffer.clear()
```

Comment:
Please add/move `self._serial_buffer.clear()` to `_clear_internal_resources`

Reply:
I will add clearing serial buffer to `_clear_internal_resources` method and remove it from `disconnect`. I found that after reconnecting, the `exec_command` method is looking for a prompt, that would be cleared from serial.
Thanks for finding it.